### DISCUSSION TITLE
Better handling of cache and config files

### DISCUSSION THREAD

**Discussion by gdubcs** [Open-Lens/app#44] [Ideas]
- only config files should be stored in `~/.config/OpenLens`
- cahe dirs ("Cache" & "Code Cache") should be tored in `~/.cache`
- Binaries should be stored in `~/.local/bin` or `~/.local/bin/open-lens`
- Move extensions data stored in `~/.k8slens` to `~/.config` ?

**Comment by fbaldo31** [comment]
Improvments are always good ideas but I see potential issues of moving every listed directories:

- We may think about users who have migrated from OpenLens to Lens but who would like to come back to OpenLens. With that changes they could experience to reinstall extensions for example, or lost their settings. But to be honest I don't know if we really have to keep full compatibility with Lens. 
- What about non Linux based OS ?
- Manage multiple locations requires to cleanup on uninstall :thinking: 

This topic may move to `Discussions` maybe because it not really an issue right ?

**Comment by gdubcs** [comment]
- To your first point, keep in mind that we have no idea what Mirantis changed in their releases since they closed the source, so I'm not sure we should burden ourselves with maintaining compatibility with Lens anyway
- Regarding non-Linux OSes, what about them? I have no idea if the problem exists to be honest
- There is no cleanup currently ; besides, `~/.config` is usually left alone on uninstall which is more of an issue in our case since that means the cache will stay forever...

I don't see how having this as an issue is problematic honestly but move it if you must

**Comment by gdubcs** [comment]
I understand this requires a lot of work but, being a very low priority, we are in no rush to fix it

**Comment by fbaldo31** [comment]
> I understand this requires a lot of work but, being a very low priority, we are in no rush to fix it

Yes that was the idea to push it under discussion.
As we have to clarify some points before just create a task and integrate it.

So for the point: What about non Linux OS ? I guess electron should manage config folder more or less like on Linux (for Windows should be under %APPDATA% but for cache I have to check what alternative the api offers.
